typecheck

```diff
@@ -1,12 +1,43 @@
 #!/bin/bash
+
+# Configuration files:
+# ====================
+#
+# * For a single program (say "program.links"), create the file
+#   "program.config" in the same directory that will contain the Links
+#   configuation to run that program.
+#
+# * For multiple programs in the same directory, you can also use a
+#   default config: "_default.config" - this is the fallback if there
+#   is no "program.config".
+#
+# * If none of these are present, Links will run with the default
+#   configuration.
+
 FAILURES=0
 for p in $(find -P . -path ./_opam -prune -false -o -name "*.links"); do
   printf "%s..." $p
   grep "^$p" "ignorelist" > /dev/null 2>&1
   if [[ $? -eq 0 ]]; then
     printf " IGNORED\n"
   else
-    linx --set=typecheck_only=true $p
+    dirname=$(dirname "$p")
+    basename=$(basename "$p")
+    filename=${basename%.*}
+    # Find an applicable config file if it exists
+    if [[ -f "$dirname/$filename.config" ]]; then
+      config="$dirname/$filename.config"
+    elif [[ -f "$dirname/_default.config" ]]; then
+      config="$dirname/_default.config"
+    else
+      config=""
+    fi
+    # Run Links, possibly with the config
+    if [[ -n $config ]]; then
+      linx --set=typecheck_only=true --config=$config $p
```

Comment:
Let's reorder the parameters here, such that `typecheck_only` is always set to `true` irrespective of the setting in `$config`.
```suggestion
      linx --config=$config --set=typecheck_only=true  $p
```

Reply:
Done :)